### PR TITLE
Add unit tests - kevin z (Login functions, basic client, small fix for password hashing)

### LINKED ISSUE #50: Starting the framework for unit testing - flask
Our flask API server needs unit testing. As part of Lab 6 efforts, the pytest framework should be used to perform unit tests on all the API functions.

### PR DESCRIPTION
**Pull request by UTkzhang**
Add unit tests - kevin z (Login functions, basic client, small fix for password hashing)

### CODE REVIEW

File: PlateUp-flask/tests/run_test.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    """Login helper function"""
+    return client.post("/login", json=dict(email=email, password=password))
+
+def logout(client):
+    """Logout helper function"""
+    return client.delete("/login", follow_redirects=True)
+
+def test_database(client):
+    """initial test. ensure that the database exists"""
+    print(Path(TEST_DB).is_file())
```
Comment (eanile) [inline PlateUp-flask/tests/run_test.py]: Might want to remove this.

File: PlateUp-flask/tests/run_test.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+################################################################################
+# Implemented by Kevin Zhang for Lab 6 - 2020/10/19
+################################################################################
+@pytest.fixture
+def client():
+    app.config["TESTING"] = True
+    app.config["SQLALCHEMY_DATABASE_URI"] = 'sqlite:///' + TEST_DB
+    db.create_all()  # setup
+    yield app.test_client()  # tests run here
```
Comment (eanile) [inline PlateUp-flask/tests/run_test.py:23]: Why is this yielded?

File: PlateUp-flask/tests/run_test.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    """Logout helper function"""
+    return client.delete("/login", follow_redirects=True)
+
+def test_database(client):
+    """initial test. ensure that the database exists"""
+    print(Path(TEST_DB).is_file())
+    tester = Path(TEST_DB).is_file()
+    assert tester
+
+def test_login_logout(client):
```
Comment (eanile) [inline PlateUp-flask/tests/run_test.py]: login/logout flow doesn't seem like a unit test to me.

**Review by eanile — CHANGES_REQUESTED** [review]
Remove __pycache__ folders. Also, is test_db.sqlite supposed to be included?

File: PlateUp-flask/tests/run_test.py (hunk 2 above)
Comment (UTkzhang) [reply PlateUp-flask/tests/run_test.py:23]: from lab sample, i believe yield here serves to safeguard memory for the yielded instance of the app

File: PlateUp-flask/tests/run_test.py (hunk 3 above)
Comment (UTkzhang) [reply PlateUp-flask/tests/run_test.py]: why not? it is 4 unit contests combined. You can't log in without logging out, you can unit these the 2 separately but the effect is the same. This is also from the lab.

File: PlateUp-flask/tests/run_test.py (hunk 1 above)
Comment (UTkzhang) [reply PlateUp-flask/tests/run_test.py]: test_database?

Comment (eanile) [reply PlateUp-flask/tests/run_test.py]: The print statement

File: PlateUp-flask/tests/run_test.py (hunk 2 above)
Comment (eanile) [reply PlateUp-flask/tests/run_test.py:23]: Ah okay, I haven't read through the lab yet.

File: PlateUp-flask/tests/run_test.py (hunk 3 above)
Comment (eanile) [reply PlateUp-flask/tests/run_test.py]: Fair enough. The last two tests in this function do seem like they could be separate tests for invalid user.

File: PlateUp-flask/tests/run_test.py (hunk 1 above)
Comment (UTkzhang) [reply PlateUp-flask/tests/run_test.py]: rmed and ignored

Comment (UTkzhang) [reply PlateUp-flask/tests/run_test.py]: fixed

File: PlateUp-flask/tests/run_test.py (hunk 3 above)
Comment (UTkzhang) [reply PlateUp-flask/tests/run_test.py]: sure, was just a lab template but i can split

Comment (UTkzhang) [reply PlateUp-flask/tests/run_test.py]: fixed

**Review by eanile — APPROVED** [review]

**Pull request merged**